bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -172,18 +206,85 @@ private void processEvent(String content) {
                     ipCameraHandler.setChannelState(CHANNEL_TOO_BLURRY_ALARM, OnOffType.OFF);
                 }
                 break;
+            case "AccessControl":
+                if ("Pulse".equals(action)) {
+                    if (content.contains("\"Method\" : 1")) {
+                        if (content.contains("\"ErrorCode\" : 0")) {
+                            startIndex = content.indexOf("CardNo", startIndex) + 11;
+                            if (startIndex > 0) {
+                                endIndex = content.indexOf(",", startIndex) - 1;
+                                String cardNo = content.substring(startIndex, endIndex);
+                                ipCameraHandler.setChannelState(CHANNEL_ACCEPTED_CARD_NUMBER, new StringType(cardNo));
+                                ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                            }
+                        }
+                    } else if (content.contains("\"Method\" : 5")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Door opened from button");
+                    } else if (content.contains("\"Method\" : 4")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Door opened remotely");
+                    }
+                } else {
+                    ipCameraHandler.logger.debug("Unrecognised Access control Dahua event, content={}", content);
+                }
+                break;
+            case "DoorCard":
+                if ("Pulse".equals(action)) {
+                    if (content.contains("\"Number\"")) {
+                        startIndex = content.indexOf("Number", startIndex) + 11;
+                        if (startIndex > 0) {
+                            endIndex = content.indexOf(",", startIndex) - 1;
+                            String cardNo = content.substring(startIndex, endIndex);
+                            ipCameraHandler.setChannelState(CHANNEL_UNACCEPTED_CARD_NUMBER, new StringType(cardNo));
+                        }
+                    }
+                } else {
+                    ipCameraHandler.logger.debug("Unrecognised Access control Dahua event, content={}", content);
+                }
+                break;
+            case "ProfileAlarmTransmit":
+                if ("Start".equals(action)) {
+                    if (content.contains("DoorMagnetism"))
+                        ipCameraHandler.setChannelState(CHANNEL_MAGNETIC_LOCK_WARNING, OnOffType.ON);
```

Comment:
Please add {} around the block: if (condition) { code ]

Reply:
Done